circuit_knitting/utils/observable_terms.py

```diff
@@ -0,0 +1,90 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2024.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+r"""Utilities for working with the unique terms of a collection of :class:`~qiskit.quantum_info.SparsePauliOp`\ s."""
+
+from __future__ import annotations
+
+from typing import Sequence, Iterable, Mapping
+
+from qiskit.quantum_info import Pauli, PauliList, SparsePauliOp
+from .iteration import strict_zip
+
+
+def gather_unique_observable_terms(
+    observables: PauliList | Sequence[Pauli | SparsePauliOp],
```

Comment:
I should swap the order here so `PauliList` gets secondary billing.

Reply:
I didn't know that the order mattered. Could you clarify what you mean here?

Reply:
Okay, I now understand what you meant.